core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java

```diff
@@ -611,12 +621,17 @@ public PublisherResult<R> finish() {
 
       @Override
       protected PublisherResult<R> addValidResponse(Address sender, ValidResponse response) {
-         PublisherResult<R> results = (PublisherResult<R>) response.getResponseValue();
-         if (log.isTraceEnabled()) {
-            log.tracef("Result was: %s for segments %s from %s with %s suspected segments", results.getResult(),
-                  targetSegments, sender, results.getSuspectedSegments());
+         Object value = response.getResponseValue();
+         if (value instanceof PublisherResult) {
+            PublisherResult<R> results = (PublisherResult<R>) value;
+            if (log.isTraceEnabled()) {
+               log.tracef("Result was: %s for segments %s from %s with %s suspected segments", results.getResult(),
+                     targetSegments, sender, results.getSuspectedSegments());
+            }
+            return results;
          }
-         return results;
+
```

Comment:
We should add a trace for when the value isn't a PublisherResult as well. Which afai can tell is only for a size return value.